codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -1568,6 +1571,9 @@ trace msg x = unsafePerformIO $ do
 reportError :: SomeException -> IO ()
 reportError e = js_reportRuntimeError True (textToJSString (pack (show e)))
 
+addMessage :: SomeException -> IO ()
+addMessage e = js_reportRuntimeError True (textToJSString (pack (C.unpack (filterOutput (C.pack (show e))))))
```

Comment:
Actually, this isn't so much what I had in mind.  I was thinking of something more like:

In `codeworld-api:CodeWorld.Internal`:

    addMessage :: Text -> IO ()

    run :: (Text -> IO ()) -> normal parameters -> IO ()
    run messageHandler params = stuff `catch` handle
      where handle exc = do
                messageHandler (pack (show e))
                setErrorState

In the `codeworld-api:CodeWorld`, things like:

    interactionOf :: normal parameters -> IO ()
    interactionOf params = run addMessage params

And in `codeworld-base`:

    interactionOf params = CodeWorld.Internal.run messageHandler params
      where messageHandler = addMessage . sanitizeMessage

Then ErrorSanitizer would be a dependency of `codeworld-base`, not `codeworld-api`.